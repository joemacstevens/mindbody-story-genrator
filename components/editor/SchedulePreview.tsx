import React from 'react';
import type { Schedule, Style, ScheduleElementId, ScheduleElementStyle } from '../../types';
import { cn } from '../../utils/cn';
import { CONTENT_ELEMENT_META, getDefaultElementStyle } from './contentElements';

type DeviceOption = 'mobile' | 'tablet' | 'desktop';

interface SchedulePreviewProps {
  schedule: Schedule;
  style: Style;
  device: DeviceOption;
  visibleElements: ScheduleElementId[];
  elementStyles: Record<ScheduleElementId, ScheduleElementStyle>;
}

const devicePadding: Record<DeviceOption, string> = {
  mobile: 'px-8 py-10',
  tablet: 'px-12 py-12',
  desktop: 'px-16 py-14',
};

const dividerClass: Record<NonNullable<Style['dividerStyle']>, string> = {
  none: 'border-none',
  thin: 'border-t border-border-light/40',
  thick: 'border-t-2 border-border-light/60',
  dotted: 'border-t border-dotted border-border-light/40',
};

const getReadableTextColor = (color: string, fallback: string): string => {
  if (!color || !color.startsWith('#')) return fallback;
  const hex = color.replace('#', '');
  const normalized = hex.length === 3 ? hex.split('').map((char) => char + char).join('') : hex.padEnd(6, '0');
  const r = parseInt(normalized.slice(0, 2), 16);
  const g = parseInt(normalized.slice(2, 4), 16);
  const b = parseInt(normalized.slice(4, 6), 16);
  if (Number.isNaN(r) || Number.isNaN(g) || Number.isNaN(b)) return fallback;

  const luminance = (0.299 * r + 0.587 * g + 0.114 * b) / 255;
  return luminance > 0.6 ? '#111827' : '#F8FAFC';
};

export const SchedulePreview: React.FC<SchedulePreviewProps> = ({
  schedule,
  style,
  device,
  visibleElements,
  elementStyles,
}) => {
  const showHeading = style.showHeading !== false;
  const showSubtitle = style.showSubtitle !== false;
  const showSchedule = style.showSchedule !== false;
  const showFooter = style.showFooter !== false;

  const hasBackgroundImage = Boolean(style.bgImage);
  const accentTextColor = getReadableTextColor(style.accent, style.backgroundColor);

  const backgroundStyles: React.CSSProperties = {
    fontFamily: style.fontFamily,
    color: style.textColorPrimary,
    backgroundColor: style.backgroundColor,
    backgroundImage: hasBackgroundImage
      ? style.overlayColor && style.overlayColor !== 'rgba(0, 0, 0, 0)'
        ? `linear-gradient(${style.overlayColor}, ${style.overlayColor}), url(${style.bgImage})`
        : `url(${style.bgImage})`
      : undefined,
    backgroundSize: style.bgFit ?? 'cover',
    backgroundPosition: style.bgPosition ?? '50% 50%',
    backgroundRepeat: 'no-repeat',
    transition: 'background-color 0.3s ease, color 0.3s ease',
  };

  return (
    <div
      className={cn(
        'relative flex h-full w-full flex-col overflow-hidden rounded-[inherit]',
        devicePadding[device],
      )}
      style={backgroundStyles}
    >
      <div className="pointer-events-none absolute inset-0 bg-[radial-gradient(circle_at_top,rgba(255,255,255,0.05),transparent_55%),radial-gradient(circle_at_bottom,rgba(139,123,216,0.08),transparent_60%)] opacity-40" />
      <div className="relative z-10 flex h-full flex-col gap-6">
        {(showHeading || showSubtitle) && (
          <header className="space-y-2 text-center" style={{ color: style.textColorPrimary }}>
            {showHeading && (
              <h1
                className="text-4xl font-bold tracking-tight sm:text-5xl"
                style={{ color: style.textColorPrimary, fontWeight: style.headingWeight ?? '700', transition: 'color 0.3s ease' }}
              >
                {style.heading}
              </h1>
            )}
            <p className="text-sm uppercase tracking-[0.4em] text-text-tertiary">{schedule.date}</p>
            {showSubtitle && (
              <p
                className="text-base text-text-secondary"
                style={{ color: style.textColorSecondary, transition: 'color 0.3s ease' }}
              >
                {style.subtitle}
              </p>
            )}
          </header>
        )}

        {style.dividerStyle && style.dividerStyle !== 'none' && (
          <div
            className={cn('opacity-80', dividerClass[style.dividerStyle])}
            style={{ borderColor: style.textColorSecondary }}
          />
        )}

        {showSchedule && (
          <main className="flex-1 space-y-6 overflow-hidden">
            {schedule.items.length === 0 ? (
              <div
                className="flex h-full flex-col items-center justify-center rounded-3xl border border-border-light/40 p-8 text-sm text-text-tertiary"
                style={{ backgroundColor: style.cardBackgroundColor }}
              >
                Add classes on the right to see them previewed here.
              </div>
            ) : (
              <ul className="grid gap-4">
                {schedule.items.map((item, index) => {
                  const renderedElements = visibleElements.reduce<
                    Array<{ id: ScheduleElementId; node: React.ReactNode }>
                  >((acc, elementId) => {
                    const meta = CONTENT_ELEMENT_META[elementId];
                    if (!meta) return acc;
                    const rawValue = item[meta.field];
                    if (!rawValue) return acc;

                    const elementStyle = elementStyles[elementId] ?? getDefaultElementStyle(elementId);
                    const fontSizePx = elementStyle.fontSize ?? meta.defaultFontSize ?? 16;
                    const fontWeight = elementStyle.fontWeight ?? meta.defaultFontWeight ?? 500;
                    const letterSpacingValue = `${(
                      elementStyle.letterSpacing ?? meta.defaultLetterSpacing ?? 0
                    ).toFixed(2)}px`;
                    const lineHeightValue = elementStyle.lineHeight ?? meta.defaultLineHeight ?? 1.3;
                    const textColor = elementStyle.color ?? meta.defaultColor ?? style.textColorSecondary;

                    let node: React.ReactNode | null = null;

                    switch (elementId) {
                      case 'time':
                        node = (
                          <div
                            className="inline-flex min-w-[88px] items-center justify-center rounded-2xl px-4 py-3 font-semibold shadow-inner"
                            style={{
                              backgroundColor: style.accent,
                              color: elementStyle.color || accentTextColor,
                              fontSize: `${fontSizePx}px`,
                              fontWeight,
                              letterSpacing: letterSpacingValue,
                              lineHeight: lineHeightValue,
                              transition:
                                'background-color 0.3s ease, color 0.3s ease, transform 0.3s ease',
                            }}
                          >
                            {rawValue}
                          </div>
                        );
                        break;
                      case 'className':
                        node = (
                          <p
                            className="font-semibold"
                            style={{
                              fontSize: `${fontSizePx}px`,
                              fontWeight,
                              letterSpacing: letterSpacingValue,
                              lineHeight: lineHeightValue,
                              color: textColor ?? style.textColorPrimary,
                              transition: 'color 0.3s ease',
                            }}
                          >
                            {rawValue}
                          </p>
                        );
                        break;
                      case 'instructor':
                        node = (
                          <p
                            style={{
                              fontSize: `${fontSizePx}px`,
                              fontWeight,
                              letterSpacing: letterSpacingValue,
                              lineHeight: lineHeightValue,
                              color: textColor ?? style.textColorSecondary,
                              transition: 'color 0.3s ease',
                            }}
                          >
                            with {rawValue}
                          </p>
                        );
                        break;
                      case 'location':
                        node = (
                          <p
                            style={{
                              fontSize: `${fontSizePx}px`,
                              fontWeight,
                              letterSpacing: letterSpacingValue,
                              lineHeight: lineHeightValue,
                              color: textColor ?? style.textColorSecondary,
                              transition: 'color 0.3s ease',
                            }}
                          >
                            üìç {rawValue}
                          </p>
                        );
                        break;
                      case 'duration':
                        node = (
                          <p
                            style={{
                              fontSize: `${fontSizePx}px`,
                              fontWeight,
                              letterSpacing: letterSpacingValue,
                              lineHeight: lineHeightValue,
                              color: textColor ?? style.textColorSecondary,
                              transition: 'color 0.3s ease',
                            }}
                          >
                            ‚è±Ô∏è {rawValue}
                          </p>
                        );
                        break;
                      case 'description':
                        node = (
                          <p
                            style={{
                              fontSize: `${fontSizePx}px`,
                              fontWeight,
                              letterSpacing: letterSpacingValue,
                              lineHeight: lineHeightValue,
                              color: textColor ?? style.textColorSecondary,
                              transition: 'color 0.3s ease',
                            }}
                          >
                            {rawValue}
                          </p>
                        );
                        break;
                      default:
                        break;
                    }

                    if (node) {
                      acc.push({ id: elementId, node });
                    }

                    return acc;
                  }, []);

                  const timeNodes = renderedElements.filter((element) => element.id === 'time');
                  const textNodes = renderedElements.filter((element) => element.id !== 'time');

                  return (
                    <li
                      key={`${item.time}-${index}`}
                      className="group flex rounded-3xl border p-5 shadow-sm transition hover:border-primary/40 hover:bg-primary/5"
                      style={{
                        backgroundColor: style.cardBackgroundColor,
                        borderColor: 'rgba(255,255,255,0.08)',
                        animation: `slide-up-fade 0.6s ease ${index * 0.08}s both`,
                        transition: 'background-color 0.3s ease, border-color 0.3s ease',
                      }}
                    >
                      <div className="flex w-full items-start gap-4">
                        {timeNodes.length > 0 ? (
                          <div className="flex flex-col gap-3">
                            {timeNodes.map((element, idx) => (
                              <React.Fragment key={`${element.id}-${idx}`}>{element.node}</React.Fragment>
                            ))}
                          </div>
                        ) : null}
                        {textNodes.length > 0 ? (
                          <div className="flex min-w-0 flex-1 flex-col gap-2">
                            {textNodes.map((element, idx) => (
                              <React.Fragment key={`${element.id}-${idx}`}>{element.node}</React.Fragment>
                            ))}
                          </div>
                        ) : null}
                      </div>
                    </li>
                  );
                })}
              </ul>
            )}
          </main>
        )}

        {showFooter && (
          <footer className="mt-auto flex flex-col gap-3 text-sm text-text-tertiary">
            {style.footerBar && (
              <div
                className="h-2 rounded-full"
                style={{ backgroundColor: style.accent, transition: 'background-color 0.3s ease' }}
              />
            )}
            <p style={{ color: style.textColorSecondary, transition: 'color 0.3s ease' }}>{style.footer}</p>
          </footer>
        )}
      </div>
    </div>
  );
};

export default SchedulePreview;
