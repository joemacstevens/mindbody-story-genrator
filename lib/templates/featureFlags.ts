export type TemplateFeatureFlags = {
  templateRegistryPreviews?: boolean;
};

declare global {
  // eslint-disable-next-line no-var
  var __STUDIOGRAM_TEMPLATE_FLAGS__: TemplateFeatureFlags | undefined;
}

const DEFAULT_FLAGS: Required<TemplateFeatureFlags> = {
  templateRegistryPreviews: true,
};

export const isTemplateRegistryPreviewEnabled = (): boolean => {
  const flags = globalThis.__STUDIOGRAM_TEMPLATE_FLAGS__;
  if (typeof flags?.templateRegistryPreviews === 'boolean') {
    return flags.templateRegistryPreviews;
  }
  return DEFAULT_FLAGS.templateRegistryPreviews;
};

/** @internal - testing hook */
export const __dangerousSetTemplateRegistryPreviewFlagForTests = (
  value: boolean | undefined,
): void => {
  const flags = (globalThis.__STUDIOGRAM_TEMPLATE_FLAGS__ ??= {});
  if (typeof value === 'boolean') {
    flags.templateRegistryPreviews = value;
  } else {
    delete flags.templateRegistryPreviews;
  }
};

export {};
