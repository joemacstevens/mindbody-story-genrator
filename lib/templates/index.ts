import { CLASSIC_TEMPLATE_DEFINITION } from './definitions/classic';
import {
  registerTemplate,
  setFallbackTemplateId,
  getTemplateDefinition,
  getTemplateDefinitionOrThrow,
  getFallbackTemplateId,
  listTemplateDefinitions,
  isTemplateRegistered,
  validateTemplateDefinition,
  __dangerousResetTemplateRegistryForTests,
  type TemplateDefinition,
  type TemplateDefaultFactories,
  type TemplateMetadata,
  type TemplateValidationIssue,
  type TemplateValidationResult,
} from './registry';

if (!isTemplateRegistered(CLASSIC_TEMPLATE_DEFINITION.id)) {
  registerTemplate(CLASSIC_TEMPLATE_DEFINITION, { fallback: true });
}

export {
  registerTemplate,
  setFallbackTemplateId,
  getTemplateDefinition,
  getTemplateDefinitionOrThrow,
  getFallbackTemplateId,
  listTemplateDefinitions,
  isTemplateRegistered,
  validateTemplateDefinition,
  __dangerousResetTemplateRegistryForTests,
};

export type {
  TemplateDefinition,
  TemplateDefaultFactories,
  TemplateMetadata,
  TemplateValidationIssue,
  TemplateValidationResult,
};

export { CLASSIC_TEMPLATE_DEFINITION, CLASSIC_TEMPLATE_ID } from './definitions/classic';
