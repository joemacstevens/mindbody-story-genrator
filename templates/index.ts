import type { TemplateId } from '../types';
import type { TemplateGalleryCategory, TemplateModule } from './types';
import { CLASSIC_TEMPLATE_MODULE, CLASSIC_TEMPLATE_ID, CLASSIC_TEMPLATE_BASE_STYLE, CLASSIC_TEMPLATE_METADATA } from './classic';

export const TEMPLATE_GALLERY_CATEGORIES: TemplateGalleryCategory[] = [
  {
    id: 'signature',
    name: 'Signature',
    description: 'Studiogram default layouts crafted for daily gym schedules.',
    icon: 'âœ¨',
  },
];

export const TEMPLATE_MODULES: TemplateModule[] = [CLASSIC_TEMPLATE_MODULE];

export const BUILT_IN_TEMPLATE_IDS = new Set<TemplateId>(
  TEMPLATE_MODULES.map((module) => module.id),
);

export const getTemplateModule = (templateId: TemplateId): TemplateModule | null =>
  TEMPLATE_MODULES.find((module) => module.id === templateId) ?? null;

export { CLASSIC_TEMPLATE_ID, CLASSIC_TEMPLATE_BASE_STYLE, CLASSIC_TEMPLATE_METADATA } from './classic';
export type { TemplateModule, TemplateGalleryCategory } from './types';
export type { TemplateGalleryCategoryId } from './types';
