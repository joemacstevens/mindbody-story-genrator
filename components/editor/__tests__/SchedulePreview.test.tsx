import React from 'react';
import { describe, it, expect, beforeEach, afterEach } from 'vitest';
import { renderToStaticMarkup } from 'react-dom/server';
import { SchedulePreview } from '../SchedulePreview';
import { MOCK_SCHEDULE } from '../../../constants';
import {
  CLASSIC_TEMPLATE_DEFINITION,
  CLASSIC_TEMPLATE_ID,
  __dangerousResetTemplateRegistryForTests,
  registerTemplate,
} from '../../../lib/templates';
import { __dangerousSetTemplateRegistryPreviewFlagForTests } from '../../../lib/templates/featureFlags';
import type { TemplateId } from '../../../types';

describe('SchedulePreview registry integration', () => {
  beforeEach(() => {
    __dangerousResetTemplateRegistryForTests();
    registerTemplate(CLASSIC_TEMPLATE_DEFINITION, { fallback: true });
    __dangerousSetTemplateRegistryPreviewFlagForTests(true);
  });

  afterEach(() => {
    __dangerousSetTemplateRegistryPreviewFlagForTests(undefined);
  });

  it('renders consistently when using registry defaults', () => {
    const definition = CLASSIC_TEMPLATE_DEFINITION;

    const manualMarkup = renderToStaticMarkup(
      <SchedulePreview
        schedule={MOCK_SCHEDULE}
        style={definition.defaults.createStyle()}
        visibleElements={definition.defaults.createVisibleElements()}
        elementStyles={definition.defaults.createElementStyles()}
        spacingScales={definition.defaults.createSmartSpacing()}
      />,
    );

    const registryMarkup = renderToStaticMarkup(
      <SchedulePreview templateId={CLASSIC_TEMPLATE_ID} schedule={MOCK_SCHEDULE} />,
    );

    expect(registryMarkup).toBe(manualMarkup);
  });

  it('matches the canonical classic template snapshot', () => {
    const markup = renderToStaticMarkup(
      <SchedulePreview templateId={CLASSIC_TEMPLATE_ID} schedule={MOCK_SCHEDULE} />,
    );

    expect(markup).toMatchSnapshot();
  });

  it('falls back to the registered default template when id is missing', () => {
    const fallbackMarkup = renderToStaticMarkup(
      <SchedulePreview templateId={CLASSIC_TEMPLATE_ID} schedule={MOCK_SCHEDULE} />,
    );

    const missingMarkup = renderToStaticMarkup(
      <SchedulePreview templateId={'missing-template' as TemplateId} schedule={MOCK_SCHEDULE} />,
    );

    expect(missingMarkup).toBe(fallbackMarkup);
  });
});
