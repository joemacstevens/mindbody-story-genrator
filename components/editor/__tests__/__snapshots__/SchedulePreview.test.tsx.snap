// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`SchedulePreview registry integration > matches the canonical classic template snapshot 1`] = `"<div class="story-canvas relative flex h-full w-full flex-col overflow-hidden rounded-[inherit] px-6 py-10" style="font-family:&#x27;Inter&#x27;, sans-serif;color:#FFFFFF;background-color:#111827;background-size:cover;background-position:50% 50%;background-repeat:no-repeat;--font-heading:48px;--font-subtitle:20px;--font-date:14px;--font-class:18px;--font-time:16px;--font-instructor:14px;--font-location:13px;--font-description:13px;--font-footer:16px;--story-hero-gap:22px;--story-schedule-gap:12px;--story-card-padding:18px;--story-footer-gap:18px;--story-time-padding-y:12px;--story-time-padding-x:18px;--story-time-min-width:80px;--story-card-accent-inset:8px;--story-card-accent-offset:8px;--story-inner-gap:16px;--story-text-gap:11px;--story-time-gap:10px;--story-logo-gap:24px;--story-logo-bottom-padding:36px"><div class="pointer-events-none absolute inset-0 z-0 bg-[radial-gradient(circle_at_top,rgba(255,255,255,0.05),transparent_55%),radial-gradient(circle_at_bottom,rgba(139,123,216,0.08),transparent_60%)] opacity-40"></div><div class="relative z-10 flex h-full flex-col"><div class="flex flex-1 flex-col" style="gap:var(--story-hero-gap)"><header class="flex flex-col items-center text-center" style="gap:var(--story-hero-gap)"><h1 class="text-story-heading font-bold tracking-tight text-center" style="color:#F8FAFC;font-weight:800;letter-spacing:-0.50px;line-height:1.1;font-size:var(--font-heading);transition:color 0.3s ease, font-weight 0.3s ease">CLASS SCHEDULE</h1><p class="text-story-date uppercase" style="color:#A5B4FC;font-weight:600;letter-spacing:2.00px;line-height:1.2;font-size:var(--font-date);transition:color 0.3s ease, font-weight 0.3s ease">Wednesday</p><p class="text-story-subtitle" style="color:#CBD5E1;font-weight:500;letter-spacing:0.00px;line-height:1.35;font-size:var(--font-subtitle);transition:color 0.3s ease, font-weight 0.3s ease">Wednesday</p></header><main class="relative flex flex-col overflow-hidden" style="flex:0 0 auto;height:65.85714285714286%;min-height:65.85714285714286%;max-height:65.85714285714286%"><ul class="flex h-full flex-1 flex-col" style="gap:var(--story-schedule-gap)"><li class="group relative flex min-h-0 flex-col border transition hover:border-primary/40 hover:bg-primary/5 flex-1 shadow-sm" style="background-color:#ef4444;border-color:rgba(255, 255, 255, 0.14);border-radius:14px;padding:var(--story-card-padding);padding-left:calc(var(--story-card-padding) + 6px);display:flex;flex-direction:column;flex:0 1 auto;min-height:0;transition:background-color 0.3s ease, border-color 0.3s ease;animation:slide-up-fade 0.6s ease 0s both"><div class="flex w-full items-start" style="gap:var(--story-inner-gap)"><div class="flex flex-col" style="gap:var(--story-time-gap)"><div class="text-story-time inline-flex items-center justify-center rounded-2xl font-semibold shadow-inner" style="color:#FFFFFF;font-weight:600;letter-spacing:0.50px;line-height:1.1;font-size:var(--font-time);transition:background-color 0.3s ease, color 0.3s ease, transform 0.3s ease;background-color:#ef4444;padding:var(--story-time-padding-y) var(--story-time-padding-x);min-width:var(--story-time-min-width)">5:10AM</div></div><div class="flex min-w-0 flex-1 flex-col" style="gap:var(--story-text-gap)"><p class="text-story-class font-semibold" style="color:#F8FAFC;font-weight:700;letter-spacing:0.00px;line-height:1.2;font-size:var(--font-class);transition:color 0.3s ease, font-weight 0.3s ease">FULL BODY</p><p class="text-story-instructor" style="color:#CBD5E1;font-weight:500;letter-spacing:0.00px;line-height:1.35;font-size:var(--font-instructor);transition:color 0.3s ease, font-weight 0.3s ease">with (Sub: Kerri Daniels)</p><p class="text-story-location" style="color:#94A3B8;font-weight:500;letter-spacing:0.00px;line-height:1.3;font-size:var(--font-location);transition:color 0.3s ease, font-weight 0.3s ease">📍 Studio A</p></div><span class="pointer-events-none absolute rounded-full" style="top:var(--story-card-accent-inset);bottom:var(--story-card-accent-inset);left:var(--story-card-accent-offset);width:3px;background-color:#ef4444;opacity:0.6"></span></div></li><li class="group relative flex min-h-0 flex-col border transition hover:border-primary/40 hover:bg-primary/5 flex-1 shadow-sm" style="background-color:#ef4444;border-color:rgba(255, 255, 255, 0.14);border-radius:14px;padding:var(--story-card-padding);padding-left:calc(var(--story-card-padding) + 6px);display:flex;flex-direction:column;flex:0 1 auto;min-height:0;transition:background-color 0.3s ease, border-color 0.3s ease;animation:slide-up-fade 0.6s ease 0.08s both"><div class="flex w-full items-start" style="gap:var(--story-inner-gap)"><div class="flex flex-col" style="gap:var(--story-time-gap)"><div class="text-story-time inline-flex items-center justify-center rounded-2xl font-semibold shadow-inner" style="color:#FFFFFF;font-weight:600;letter-spacing:0.50px;line-height:1.1;font-size:var(--font-time);transition:background-color 0.3s ease, color 0.3s ease, transform 0.3s ease;background-color:#ef4444;padding:var(--story-time-padding-y) var(--story-time-padding-x);min-width:var(--story-time-min-width)">6:05AM</div></div><div class="flex min-w-0 flex-1 flex-col" style="gap:var(--story-text-gap)"><p class="text-story-class font-semibold" style="color:#F8FAFC;font-weight:700;letter-spacing:0.00px;line-height:1.2;font-size:var(--font-class);transition:color 0.3s ease, font-weight 0.3s ease">FULL BODY</p><p class="text-story-instructor" style="color:#CBD5E1;font-weight:500;letter-spacing:0.00px;line-height:1.35;font-size:var(--font-instructor);transition:color 0.3s ease, font-weight 0.3s ease">with (Sub: Kerri Daniels)</p><p class="text-story-location" style="color:#94A3B8;font-weight:500;letter-spacing:0.00px;line-height:1.3;font-size:var(--font-location);transition:color 0.3s ease, font-weight 0.3s ease">📍 Studio A</p></div><span class="pointer-events-none absolute rounded-full" style="top:var(--story-card-accent-inset);bottom:var(--story-card-accent-inset);left:var(--story-card-accent-offset);width:3px;background-color:#ef4444;opacity:0.6"></span></div></li><li class="group relative flex min-h-0 flex-col border transition hover:border-primary/40 hover:bg-primary/5 flex-1 shadow-sm" style="background-color:#ef4444;border-color:rgba(255, 255, 255, 0.14);border-radius:14px;padding:var(--story-card-padding);padding-left:calc(var(--story-card-padding) + 6px);display:flex;flex-direction:column;flex:0 1 auto;min-height:0;transition:background-color 0.3s ease, border-color 0.3s ease;animation:slide-up-fade 0.6s ease 0.16s both"><div class="flex w-full items-start" style="gap:var(--story-inner-gap)"><div class="flex flex-col" style="gap:var(--story-time-gap)"><div class="text-story-time inline-flex items-center justify-center rounded-2xl font-semibold shadow-inner" style="color:#FFFFFF;font-weight:600;letter-spacing:0.50px;line-height:1.1;font-size:var(--font-time);transition:background-color 0.3s ease, color 0.3s ease, transform 0.3s ease;background-color:#ef4444;padding:var(--story-time-padding-y) var(--story-time-padding-x);min-width:var(--story-time-min-width)">9:00AM</div></div><div class="flex min-w-0 flex-1 flex-col" style="gap:var(--story-text-gap)"><p class="text-story-class font-semibold" style="color:#F8FAFC;font-weight:700;letter-spacing:0.00px;line-height:1.2;font-size:var(--font-class);transition:color 0.3s ease, font-weight 0.3s ease">FEMME BOOTCAMP</p><p class="text-story-instructor" style="color:#CBD5E1;font-weight:500;letter-spacing:0.00px;line-height:1.35;font-size:var(--font-instructor);transition:color 0.3s ease, font-weight 0.3s ease">with (Glenda Ortiz)</p><p class="text-story-location" style="color:#94A3B8;font-weight:500;letter-spacing:0.00px;line-height:1.3;font-size:var(--font-location);transition:color 0.3s ease, font-weight 0.3s ease">📍 Main Room</p></div><span class="pointer-events-none absolute rounded-full" style="top:var(--story-card-accent-inset);bottom:var(--story-card-accent-inset);left:var(--story-card-accent-offset);width:3px;background-color:#ef4444;opacity:0.6"></span></div></li><li class="group relative flex min-h-0 flex-col border transition hover:border-primary/40 hover:bg-primary/5 flex-1 shadow-sm" style="background-color:#ef4444;border-color:rgba(255, 255, 255, 0.14);border-radius:14px;padding:var(--story-card-padding);padding-left:calc(var(--story-card-padding) + 6px);display:flex;flex-direction:column;flex:0 1 auto;min-height:0;transition:background-color 0.3s ease, border-color 0.3s ease;animation:slide-up-fade 0.6s ease 0.24s both"><div class="flex w-full items-start" style="gap:var(--story-inner-gap)"><div class="flex flex-col" style="gap:var(--story-time-gap)"><div class="text-story-time inline-flex items-center justify-center rounded-2xl font-semibold shadow-inner" style="color:#FFFFFF;font-weight:600;letter-spacing:0.50px;line-height:1.1;font-size:var(--font-time);transition:background-color 0.3s ease, color 0.3s ease, transform 0.3s ease;background-color:#ef4444;padding:var(--story-time-padding-y) var(--story-time-padding-x);min-width:var(--story-time-min-width)">12:00PM</div></div><div class="flex min-w-0 flex-1 flex-col" style="gap:var(--story-text-gap)"><p class="text-story-class font-semibold" style="color:#F8FAFC;font-weight:700;letter-spacing:0.00px;line-height:1.2;font-size:var(--font-class);transition:color 0.3s ease, font-weight 0.3s ease">ZUMBA</p><p class="text-story-instructor" style="color:#CBD5E1;font-weight:500;letter-spacing:0.00px;line-height:1.35;font-size:var(--font-instructor);transition:color 0.3s ease, font-weight 0.3s ease">with (Lama Alassil)</p><p class="text-story-location" style="color:#94A3B8;font-weight:500;letter-spacing:0.00px;line-height:1.3;font-size:var(--font-location);transition:color 0.3s ease, font-weight 0.3s ease">📍 Studio B</p></div><span class="pointer-events-none absolute rounded-full" style="top:var(--story-card-accent-inset);bottom:var(--story-card-accent-inset);left:var(--story-card-accent-offset);width:3px;background-color:#ef4444;opacity:0.6"></span></div></li><li class="group relative flex min-h-0 flex-col border transition hover:border-primary/40 hover:bg-primary/5 flex-1 shadow-sm" style="background-color:#ef4444;border-color:rgba(255, 255, 255, 0.14);border-radius:14px;padding:var(--story-card-padding);padding-left:calc(var(--story-card-padding) + 6px);display:flex;flex-direction:column;flex:0 1 auto;min-height:0;transition:background-color 0.3s ease, border-color 0.3s ease;animation:slide-up-fade 0.6s ease 0.32s both"><div class="flex w-full items-start" style="gap:var(--story-inner-gap)"><div class="flex flex-col" style="gap:var(--story-time-gap)"><div class="text-story-time inline-flex items-center justify-center rounded-2xl font-semibold shadow-inner" style="color:#FFFFFF;font-weight:600;letter-spacing:0.50px;line-height:1.1;font-size:var(--font-time);transition:background-color 0.3s ease, color 0.3s ease, transform 0.3s ease;background-color:#ef4444;padding:var(--story-time-padding-y) var(--story-time-padding-x);min-width:var(--story-time-min-width)">5:15PM</div></div><div class="flex min-w-0 flex-1 flex-col" style="gap:var(--story-text-gap)"><p class="text-story-class font-semibold" style="color:#F8FAFC;font-weight:700;letter-spacing:0.00px;line-height:1.2;font-size:var(--font-class);transition:color 0.3s ease, font-weight 0.3s ease">ABS &amp; BOOTY</p><p class="text-story-instructor" style="color:#CBD5E1;font-weight:500;letter-spacing:0.00px;line-height:1.35;font-size:var(--font-instructor);transition:color 0.3s ease, font-weight 0.3s ease">with (Alexis Mobley)</p><p class="text-story-location" style="color:#94A3B8;font-weight:500;letter-spacing:0.00px;line-height:1.3;font-size:var(--font-location);transition:color 0.3s ease, font-weight 0.3s ease">📍 Strength Zone</p></div><span class="pointer-events-none absolute rounded-full" style="top:var(--story-card-accent-inset);bottom:var(--story-card-accent-inset);left:var(--story-card-accent-offset);width:3px;background-color:#ef4444;opacity:0.6"></span></div></li><li class="group relative flex min-h-0 flex-col border transition hover:border-primary/40 hover:bg-primary/5 flex-1 shadow-sm" style="background-color:#ef4444;border-color:rgba(255, 255, 255, 0.14);border-radius:14px;padding:var(--story-card-padding);padding-left:calc(var(--story-card-padding) + 6px);display:flex;flex-direction:column;flex:0 1 auto;min-height:0;transition:background-color 0.3s ease, border-color 0.3s ease;animation:slide-up-fade 0.6s ease 0.4s both"><div class="flex w-full items-start" style="gap:var(--story-inner-gap)"><div class="flex flex-col" style="gap:var(--story-time-gap)"><div class="text-story-time inline-flex items-center justify-center rounded-2xl font-semibold shadow-inner" style="color:#FFFFFF;font-weight:600;letter-spacing:0.50px;line-height:1.1;font-size:var(--font-time);transition:background-color 0.3s ease, color 0.3s ease, transform 0.3s ease;background-color:#ef4444;padding:var(--story-time-padding-y) var(--story-time-padding-x);min-width:var(--story-time-min-width)">6:15PM</div></div><div class="flex min-w-0 flex-1 flex-col" style="gap:var(--story-text-gap)"><p class="text-story-class font-semibold" style="color:#F8FAFC;font-weight:700;letter-spacing:0.00px;line-height:1.2;font-size:var(--font-class);transition:color 0.3s ease, font-weight 0.3s ease">STEP UP</p><p class="text-story-instructor" style="color:#CBD5E1;font-weight:500;letter-spacing:0.00px;line-height:1.35;font-size:var(--font-instructor);transition:color 0.3s ease, font-weight 0.3s ease">with (Audrey Carryl)</p><p class="text-story-location" style="color:#94A3B8;font-weight:500;letter-spacing:0.00px;line-height:1.3;font-size:var(--font-location);transition:color 0.3s ease, font-weight 0.3s ease">📍 Studio C</p></div><span class="pointer-events-none absolute rounded-full" style="top:var(--story-card-accent-inset);bottom:var(--story-card-accent-inset);left:var(--story-card-accent-offset);width:3px;background-color:#ef4444;opacity:0.6"></span></div></li></ul></main><footer class="mt-auto flex flex-col text-sm text-text-tertiary" style="gap:var(--story-footer-gap)"><p class="text-story-footer" style="color:#94A3B8;font-weight:500;letter-spacing:0.00px;line-height:1.4;font-size:var(--font-footer);transition:color 0.3s ease, font-weight 0.3s ease">@yourgymname</p></footer></div></div></div>"`;
